hydra/app/controllers/catalog_controller.rb

```diff
@@ -253,4 +261,14 @@ def policies
   def contributingcollections
     render "contributingcollections.html.erb"
   end
+
+  private
+
+    # Fetches the image files for each document
+    def fetch_image_files
+      @image_files = @document_list.each_with_object({}) do |document, hash|
+        acda_record = Acda.where(id: document.id).first
```

Comment:
This is a very non-performant query.  We're hitting SOLR one additional time per item in the document list.  We should instead be writing a more customized query that takes the IDs from the document_list and passes those as a single query to Solr.  Also we only need the `id` and `image_file` attributes from the sole document.

[This logic provides some guidance](https://github.com/samvera-labs/bulkrax/blob/657c6feaae8d81fe9487dbd16dc42730637b9313/app/parsers/bulkrax/parser_export_record_set.rb#L174-L183).  I would also consider extracting this logic into a class; because what you are writing in the controller is a pain to test in a controller.

Reply:
Good call!  This lead me to an entirely different solution.  Ultimately we just wanted to check if the image_file (or thumbnail_file) exists on the record.  So now i'm indexing that so we only need to interact with the solr doc

```rb
solr_doc['has_image_file_bsi'] = object.image_file.present?
solr_doc['has_thumbnail_file_bsi'] = object.thumbnail_file.present?
```